src/commcare_cloud/commands/ansible/helpers.py

```diff
@@ -35,13 +35,16 @@ def _build_env(self, args):
         return env
 
 
-def get_common_ssh_args(environment, use_pem=False):
+def get_common_ssh_args(environment, use_factory_auth=False):
     strict_host_key_checking = environment.public_vars.get('commcare_cloud_strict_host_key_checking', True)
 
     common_ssh_args = []
-    if use_pem:
-        pem = environment.public_vars.get('commcare_cloud_pem', None)
-        if pem:
+
+    if use_factory_auth:
+        if not environment.public_vars.get('commcare_cloud_pem'):
+            common_ssh_args += ('--ask-pass',)
```

Comment:
I think this won't work because `--ask-pass` isn't an ssh arg, but rather an ansible arg. All this stuff gets shell-quoted and passed together to the ansible arg `--ssh-common-args`, so I think ansible won't see that you used `--ask-pass`.

Reply:
Given that `use_factory_auth` spans spans ssh and non-ssh arguments, I might break it out to another place. You  may want to also test whether you can pass in `--ssh-common-args` more than once, since that'll affect the implementation.

Reply:
https://github.com/dimagi/commcare-cloud/pull/1702/commits/ce8f40d15e52574a8bcad6beaf0f311cd9299b67

There's some redundant code between the two places where this function gets called that I'd like to consolidate once everything is working